src/main/java/com/risksense/converters/Impl/XMLJSONConverterImpl.java

```diff
@@ -0,0 +1,148 @@
+package com.risksense.converters.Impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.risksense.converters.XMLJSONConverterI;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * @author pediredla
+ */
+public class XMLJSONConverterImpl implements XMLJSONConverterI {
+
+    /**
+     * XML implementation of the converter
+     *
+     *
+     * @param json {@link java.io.File} from which to read JSON data.
+     * @param xml  {@link java.io.File} from which to write XML data.
+     * @throws IOException
+     */
+    @Override
+    public void convertJSONtoXML(File json, File xml) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> data = mapper.readValue(json, Map.class);
+        StringBuilder sb = new StringBuilder();
+        sb.append("<object>");
+        Iterator it = data.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry pair = (Map.Entry) it.next();
+            fillData(pair, sb, mapper);
+            it.remove();
+        }
+        sb.append("</object>");
+        final String content = sb.toString();
+        FileWriter fw = new FileWriter(xml.getAbsoluteFile());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced in statements at the following lines: 43, 44. The resource closure statement is at line: 45. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Writer.write. Either a) close *fw* in a try-finally block or b) close the resource by declaring *fw* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).